### ISSUE TITLE
No module named 'gitdb.utils.compat'.

### ISSUE THREAD

**Issue by Blue-Eagle-10** [NervanaSystems/distiller#475]
I installed all the environments and packages according to the tutorial and got this error when running the code: ModuleNotFoundError: No module named 'gitdb.utils.compat'.
Ask for help, how can I resolve this error? thank you very much!

**Comment by Blue-Eagle-10** [comment]
file:///home/jiechen/%E5%9B%BE%E7%89%87/2020-02-22%2007-10-17.png

**Comment by Blue-Eagle-10** [comment]
This is a screenshot of the operation of the error part

**Comment by Blue-Eagle-10** [comment]
![2020-02-22 07-10-17](https://user-images.githubusercontent.com/24905967/75047072-25756400-5501-11ea-8f38-99d437c270c6.png)

**Comment by netaz** [comment]
Hi @Blue-Eagle-10,

You seem to be using a virtual-env, but the error shows that your setup is looking for packages outside the virtual-env.  I think you did something wrong during the installation.

Delete the virtual environment, install again and check for the problem.

Cheers,
Neta

**Comment by Blue-Eagle-10** [comment]
Hi@netaz,
Thank you for your reply. I have reinstalled the virtual environment, but unfortunately, the same error still exists. Is it a problem with the path setting? How can I solve it? Looking forward to your help！thank you very much！
![2020-02-22 17-35-03](https://user-images.githubusercontent.com/24905967/75083860-ebcf4800-5556-11ea-8415-0e0dbb7cb4df.png)
![2020-02-22 17-38-23](https://user-images.githubusercontent.com/24905967/75083874-20430400-5557-11ea-936b-eeebf0828826.png)

**Comment by Blue-Eagle-10** [comment]
> Hi @Blue-Eagle-10,
> 
> You seem to be using a virtual-env, but the error shows that your setup is looking for packages outside the virtual-env. I think you did something wrong during the installation.
> 
> Delete the virtual environment, install again and check for the problem.
> If you did everything correctly you should see:
> 
> ```
> $ ll <path-to-your-virt-env>/lib/python3.6/site-packages/gitdb/utils/compat.py
> -rw-rw-r-- 1 cvds_lab cvds_lab 1093 Jul 31  2019 <path-to-your-virt-env>/lib/python3.6/site-packages/gitdb/utils/compat.py
> ```
> 
> Cheers,
> Neta

**Comment by netaz** [comment]
Hi @Blue-Eagle-10,

I am pretty sure I found the problem.  This is a in package `gitpython`.
Try uninstalling `gitdb2` and installing it again, but forcing version 2.0.6:
```
$ pip uninstall gitdb2
$ pip install gitdb2==2.0.6
```

Long explanation (or why I think this is the problem):
- First I saw that your version of `gitdb2` is different than mine.
- Because you installed the packages from our `requirements.txt` file, and `gitdb2` is not explicitly listed in this file, I knew that some other package was dependent on `gitdb2`.
- To understand the package dependencies, I installed `pipdeptree` and ran it.
```
$ pip install pipdeptree
(env_remote_distiller_pt_1.3) $ pipdeptree | grep git
   - gitpython [required: ==2.1.11, installed: 2.1.11]
       - gitdb2 [required: >=2.0.0, installed: 2.0.6]    
```

- So `gitpython` requires `gitdb2`, version >=2.0.0.  In my case, `gitdb2` has version 2.0.6 because I installed it a long time ago.  In your case `gitdb2` has version 3.0.2
- Let's examine the source code of each of these versions:
  https://github.com/gitpython-developers/gitdb/tree/2.0.6/gitdb/utils

![image](https://user-images.githubusercontent.com/229801/75090145-afa4e300-5568-11ea-9abe-1d8f22fa2326.png)

  https://github.com/gitpython-developers/gitdb/tree/3.0.2/gitdb/utils
![image](https://user-images.githubusercontent.com/229801/75090158-c21f1c80-5568-11ea-8e0b-da35369d7cef.png)

==> This clearly shows that file `gitdb/utils/compat.py` was removed between the two versions.
This broke `gitpython` which only has a semi-definite version requirement from `gitdb2`.
So the bug is in the dependency requirements of `gitpython`.

Cheers
Neta

**Comment by Blue-Eagle-10** [comment]
Hi @Neta，

Thank you very much for your reply. I have resolved this issue according to your comments. Thank you very much for helping me solve the problem that has troubled me for so long. In addition, I think the developers of this project should take this issue as a note in the installation instructions, which will remind those who use this project in the future to pay attention to this matter! Thanks from the heart! 

sincerely，
blueeagle

**Comment by nzmora** [comment]
@levzlotnik - please note that we need to test adding to `requirements.txt` the line `gitdb2==2.0.6`, before the specification of `gitpython`.   

This will prevent others from experiencing this issue.
Thanks!
Neta

**Comment by levzlotnik** [comment]
Hi @Blue-Eagle-10  
I've tested and recreated your problem.
Relaxing the restriction on `gitpython==2.1.11` to just `gitpython` seems to solve the problem, without specifying `gitdb2` in the requirements.  
I'll post a commit which updates the requirements.

**Comment by Blue-Eagle-10** [comment]
Hi@levzlotnik,

Thank you for your reply, and your careful and detailed help, so that I can use your sharing for further research. Thank you very much for the project you shared. It is really amazing.

sincerely，
blueeagle

**Comment by Harmon758** [comment]
See See https://github.com/gitpython-developers/GitPython/issues/983..

**Comment by guyjacob** [comment]
Thanks @Harmon758. So it appears that the gitdb issue is resolved in gitpython version 3.1.0. I pushed another update to our `requirements.txt`. Closing this now.
